internal/providers/terraform/azure/function_app.go

```diff
@@ -0,0 +1,188 @@
+		costComponents = append(costComponents, AppFunctionConsumptionExecutionTimeCostComponent(executions, executionTime, memorySize, location, meterNameExecTime))
+		costComponents = append(costComponents, AppFunctionConsumptionExecutionsCostComponent(executions, location, meterNameExecutions))
+	}
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+
+func AppFunctionPremiumCPUCostComponent(instances float64, skuCPU int, location, meterName string) *schema.CostComponent {
+
+	return &schema.CostComponent{
+
+		Name:           "vCPU",
+		Unit:           "vCPU-hours",
+		UnitMultiplier: 1,
+		HourlyQuantity: decimalPtr(decimal.NewFromFloat(instances * float64(skuCPU))),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(location),
+			Service:       strPtr("Functions"),
+			ProductFamily: strPtr("Compute"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "meterName", Value: strPtr(meterName)},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr("Consumption"),
+		},
+	}
+
+}
+func AppFunctionPremiumMemoryCostComponent(instances, skuMemory float64, location, meterName string) *schema.CostComponent {
+	return &schema.CostComponent{
+
+		Name:           "Memory",
+		Unit:           "GB-hours",
+		UnitMultiplier: 1,
+		HourlyQuantity: decimalPtr(decimal.NewFromFloat(instances * skuMemory)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(location),
+			Service:       strPtr("Functions"),
+			ProductFamily: strPtr("Compute"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "meterName", Value: strPtr(meterName)},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr("Consumption"),
+		},
+	}
+}
+func AppFunctionConsumptionExecutionTimeCostComponent(executions, executionTime float64, memorySize float64, location, meterName string) *schema.CostComponent {
+	return &schema.CostComponent{
+
+		Name:           "Execution time",
+		Unit:           "GB Seconds",
+		UnitMultiplier: 1,
+		HourlyQuantity: decimalPtr(decimal.NewFromFloat(executionTime * memorySize)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(location),
+			Service:       strPtr("Functions"),
+			ProductFamily: strPtr("Compute"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "meterName", Value: strPtr(meterName)},
+				{Key: "skuName", Value: strPtr("Standard")},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr("Consumption"),
+		},
+	}
+}
+func AppFunctionConsumptionExecutionsCostComponent(executions float64, location, meterName string) *schema.CostComponent {
+	return &schema.CostComponent{
+
+		Name:            "Executions",
+		Unit:            "requests",
+		UnitMultiplier:  1,
+		MonthlyQuantity: decimalPtr(decimal.NewFromFloat(executions / 10)),
```

Comment:
@aliscott / @tim775 could we allow UnitMultiplier to be a float to address the following issue? I've only seen a few resources have this issue so it's fine if we don't change anything just now...

Azure's pricing page mentions "$0.20 per million executions", however, their pricing API returns `"unit" : "10", "USD" : "0.000002"` (i.e. they return the price per 10 requests) 🤦 

So when I enter 1,000,000 in the usage file, I see the following, the price and monthly cost is correct, but I entered 1M requests not 100K.

```
 └─ Executions                                 $0.000002      100,000  requests          $0.20
```

If we use UnitMultiplier 100,000 (as that's how we can represent the Azure pricing API figure in nicer terms), then we see the following, the price and costs map to what the Azure pricing page shows but I entered 1M...
```
 └─ Executions                                 $0.20            1  100k requests        $0.20
```

@NeverovDS no action needed for this comment.

Reply:
Wouldn't we want to set `UnitMultiplier = 100k`, `Unit = 1M requests`?

As a side note, I think the unit here should be "executions" since that's what Azure show on their pricing page.

Reply:
@aliscott that shows the following, which also seems confusing despite showing the correct price/cost:
```
 └─ Executions                                 $0.20            1  100k 1M requests        $0.20
```

Regarding the units, Azure's pricing calculator shows "requests". I went with that as it can help distinguish requests to executions from the "Execution time" cost component.

Reply:
Okay, I think with @tim775's changes in https://github.com/infracost/infracost/pull/622 then `UnitMultiplier = 100k`, `Unit = 1M requests` will work as expected, but will let him confirm.

Reply:
@NeverovDS in that case plz use `UnitMultiplier = 100000` and `Unit = 1M requests` in this PR